## Part 1: Explore the Data

Import the data and use Pandas to learn more about the dataset.

In [18]:
import pandas as pd

df = pd.read_csv('Resources/client_dataset.csv')
df

first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year         item_id     category  \
0              17        2023  EUD29711-63-6U        decor   
1              20        2023  XDA18116-89-4A  consumables   
2               4        2023  ABE59463-05-7E     software   
3              17        2023  ZMM00836-65-0C  consumables   
4              21        2023  BZX55559-12-3X  consumables   
...           ...         ...             ...          ...   
54634          17        2023  UCA28532-31-0B     software   
54635           6        2023  EBD29277-23-7U  consumables   
54636           9        2023  EBD30093-55-9Z  consumables   
54637           3        2023  XMZ38239-49-3D    furniture   
54638           1        2023  ZCZ05589-50-5D     software   

              subcategory  unit_price  unit_cost  unit_weight   qty  \
0                wall art     1096.80     762.71         7.50   105   
1                    pens       24.95      15.09         1.49    21   
2      project management       13.52       7.86         1.68    39   
3                    pens       36.42      24.85         1.23    29   
4                    misc      195.10     108.17        46.43    20   
...                   ...         ...        ...          ...   ...   
54634           logistics       83.13      51.60         2.25    33   
54635    kitchen supplies      206.59     175.46        11.70    47   
54636   bathroom supplies       65.66      57.31         4.16   475   
54637              stands        1.48       1.22        18.04   112   
54638  project management        3.01       1.61         2.07  1031   

       line_number  
0                1  
1                0  
2                6  
3                3  
4                1  
...            ...  
54634            8  
54635            0  
54636            6  
54637            0  
54638            5  

[54639 rows x 18 columns]

In [3]:
# View the column names in the data
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number'],
      dtype='object')

In [4]:
# Use the describe function to gather some basic statistics
df.describe()

client_id      order_id    order_week    order_year    unit_price  \
count  54639.000000  5.463900e+04  54639.000000  54639.000000  54639.000000   
mean   54837.869416  5.470190e+06     11.359139   2022.993064    136.267207   
std    25487.438231  2.599807e+06      7.023499      0.082997    183.873135   
min    10033.000000  1.000886e+06      1.000000   2022.000000      0.010000   
25%    33593.000000  3.196372e+06      6.000000   2023.000000     20.800000   
50%    53305.000000  5.496966e+06     11.000000   2023.000000     68.310000   
75%    78498.000000  7.733869e+06     17.000000   2023.000000    173.160000   
max    99984.000000  9.998480e+06     52.000000   2023.000000   1396.230000   

          unit_cost   unit_weight           qty   line_number  
count  54639.000000  54639.000000  5.463900e+04  54639.000000  
mean      99.446073      5.004116  5.702646e+02      2.979667  
std      133.164267      5.326599  1.879552e+04      2.436320  
min        0.010000      0.000000  0.000000e+00      0.000000  
25%       14.840000      1.450000  3.200000e+01      1.000000  
50%       49.890000      3.240000  6.800000e+01      3.000000  
75%      125.570000      6.890000  1.700000e+02      5.000000  
max      846.270000     46.430000  3.958244e+06      9.000000

In [5]:
# Use this space to do any additional research
# and familiarize yourself with the data.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54639 entries, 0 to 54638
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   first        54639 non-null  object 
 1   last         54639 non-null  object 
 2   job          54639 non-null  object 
 3   phone        54639 non-null  object 
 4   email        54639 non-null  object 
 5   client_id    54639 non-null  int64  
 6   order_id     54639 non-null  int64  
 7   order_date   54639 non-null  object 
 8   order_week   54639 non-null  int64  
 9   order_year   54639 non-null  int64  
 10  item_id      54639 non-null  object 
 11  category     54639 non-null  object 
 12  subcategory  54639 non-null  object 
 13  unit_price   54639 non-null  float64
 14  unit_cost    54639 non-null  float64
 15  unit_weight  54639 non-null  float64
 16  qty          54639 non-null  int64  
 17  line_number  54639 non-null  int64  
dtypes: float64(3), int64(6), object(9)
memory usag

In [9]:
# What three item categories had the most entries?
top3 = df['category'].value_counts().head(3)
top3

category
consumables    23538
furniture      11915
software        8400
Name: count, dtype: int64

In [11]:
# For the category with the most entries, which subcategory had the most entries?
topCat = top3.index[0]
df[df['category'] == topCat]['subcategory'].value_counts().head(1)

subcategory
bathroom supplies    6424
Name: count, dtype: int64

In [13]:
# Which five clients had the most entries in the data?
top5_clients = df['client_id'].value_counts().head(5)
top5_clients

client_id
33615    220
66037    211
46820    209
38378    207
24741    207
Name: count, dtype: int64

In [16]:
# Store the client ids of those top 5 clients in a list.
top5_clients_list = top5_clients.index.tolist()
top5_clients_list

[33615, 66037, 46820, 38378, 24741]

In [166]:
df[df['client_id']==24741].head(50)

first     last                     job         phone  \
416    Kendra  Garrett  Pension scheme manager  808.257.2029   
520    Kendra  Garrett  Pension scheme manager  808.257.2029   
671    Kendra  Garrett  Pension scheme manager  808.257.2029   
872    Kendra  Garrett  Pension scheme manager  808.257.2029   
1460   Kendra  Garrett  Pension scheme manager  808.257.2029   
1532   Kendra  Garrett  Pension scheme manager  808.257.2029   
1611   Kendra  Garrett  Pension scheme manager  808.257.2029   
1877   Kendra  Garrett  Pension scheme manager  808.257.2029   
2221   Kendra  Garrett  Pension scheme manager  808.257.2029   
2267   Kendra  Garrett  Pension scheme manager  808.257.2029   
2415   Kendra  Garrett  Pension scheme manager  808.257.2029   
3185   Kendra  Garrett  Pension scheme manager  808.257.2029   
3280   Kendra  Garrett  Pension scheme manager  808.257.2029   
3705   Kendra  Garrett  Pension scheme manager  808.257.2029   
3742   Kendra  Garrett  Pension scheme manager  808.257.2029   
4063   Kendra  Garrett  Pension scheme manager  808.257.2029   
4759   Kendra  Garrett  Pension scheme manager  808.257.2029   
4898   Kendra  Garrett  Pension scheme manager  808.257.2029   
4993   Kendra  Garrett  Pension scheme manager  808.257.2029   
5330   Kendra  Garrett  Pension scheme manager  808.257.2029   
5462   Kendra  Garrett  Pension scheme manager  808.257.2029   
5627   Kendra  Garrett  Pension scheme manager  808.257.2029   
6139   Kendra  Garrett  Pension scheme manager  808.257.2029   
6186   Kendra  Garrett  Pension scheme manager  808.257.2029   
6380   Kendra  Garrett  Pension scheme manager  808.257.2029   
6686   Kendra  Garrett  Pension scheme manager  808.257.2029   
7010   Kendra  Garrett  Pension scheme manager  808.257.2029   
7181   Kendra  Garrett  Pension scheme manager  808.257.2029   
7210   Kendra  Garrett  Pension scheme manager  808.257.2029   
7366   Kendra  Garrett  Pension scheme manager  808.257.2029   
7409   Kendra  Garrett  Pension scheme manager  808.257.2029   
7712   Kendra  Garrett  Pension scheme manager  808.257.2029   
7735   Kendra  Garrett  Pension scheme manager  808.257.2029   
8479   Kendra  Garrett  Pension scheme manager  808.257.2029   
8653   Kendra  Garrett  Pension scheme manager  808.257.2029   
9479   Kendra  Garrett  Pension scheme manager  808.257.2029   
10176  Kendra  Garrett  Pension scheme manager  808.257.2029   
10425  Kendra  Garrett  Pension scheme manager  808.257.2029   
10460  Kendra  Garrett  Pension scheme manager  808.257.2029   
10610  Kendra  Garrett  Pension scheme manager  808.257.2029   
11087  Kendra  Garrett  Pension scheme manager  808.257.2029   
11198  Kendra  Garrett  Pension scheme manager  808.257.2029   
12133  Kendra  Garrett  Pension scheme manager  808.257.2029   
12152  Kendra  Garrett  Pension scheme manager  808.257.2029   
12330  Kendra  Garrett  Pension scheme manager  808.257.2029   
12505  Kendra  Garrett  Pension scheme manager  808.257.2029   
12719  Kendra  Garrett  Pension scheme manager  808.257.2029   
13217  Kendra  Garrett  Pension scheme manager  808.257.2029   
13451  Kendra  Garrett  Pension scheme manager  808.257.2029   
13470  Kendra  Garrett  Pension scheme manager  808.257.2029   

                                 email  client_id  order_id  order_date  \
416    garrettkendra8479@jefferson.com      24741   3699423  2023-05-18   
520    garrettkendra8479@jefferson.com      24741   7980819  2023-05-25   
671    garrettkendra8479@jefferson.com      24741   9538464  2023-03-28   
872    garrettkendra8479@jefferson.com      24741   2118797  2023-02-05   
1460   garrettkendra8479@jefferson.com      24741   7565274  2023-03-21   
1532   garrettkendra8479@jefferson.com      24741   3335711  2023-04-15   
1611   garrettkendra8479@jefferson.com      24741   9114777  2023-02-13   
1877   garrettkendra8479@jefferson.com      24741   2118797  2023-02-05   
2221   garrettkendra8479@jefferson.com      24741   3699423  2023-05

In [17]:
# How many total units (the qty column) did the client with the most entries order order?
most_ent = top5_clients_list[0]

df[df['client_id'] == most_ent]['qty'].sum()

64313

## Part 2: Transform the Data
Do we know that this client spent the more money than client 66037? If not, how would we find out? Transform the data using the steps below to prepare it for analysis.

In [22]:
# Create a column that calculates the subtotal for each line using the unit_price and the qty
df['subtotal'] = df['unit_price'] * df['qty']
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year         item_id     category         subcategory  \
0          17        2023  EUD29711-63-6U        decor            wall art   
1          20        2023  XDA18116-89-4A  consumables                pens   
2           4        2023  ABE59463-05-7E     software  project management   
3          17        2023  ZMM00836-65-0C  consumables                pens   
4          21        2023  BZX55559-12-3X  consumables                misc   

   unit_price  unit_cost  unit_weight  qty  line_number   subtotal  
0     1096.80     762.71         7.50  105            1  115164.00  
1       24.95      15.09         1.49   21            0     523.95  
2       13.52       7.86         1.68   39            6     527.28  
3       36.42      24.85         1.23   29            3    1056.18  
4      195.10     108.17        46.43   20            1    3902.00

In [36]:
# Create a column for shipping price.
# Assume a shipping price of $7 per pound for orders over 50 pounds and $10 per pound for items 50 pounds or under.
# gonna use lambda for this
df['total_weight'] = df['unit_weight'] * df['qty']
df['shipping_price'] = df['total_weight'].apply(lambda x: x * 7 if x>50 else x*10)
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ... unit_cost unit_weight  qty  line_number  \
0          17        2023  ...    762.71        7.50  105            1   
1          20        2023  ...     15.09        1.49   21            0   
2           4        2023  ...      7.86        1.68   39            6   
3          17        2023  ...     24.85        1.23   29            3   
4          21        2023  ...    108.17       46.43   20            1   

    subtotal  shipping_price  line_cost  Line_price  line_profit  total_weight  
0  115164.00         5512.50   80159.55   125898.61     45739.06        787.50  
1     523.95          312.90     331.79      588.69       256.90         31.29  
2     527.28          458.64     323.34      594.41       271.07         65.52  
3    1056.18          356.70     732.95     1167.31       434.36         35.67  
4    3902.00         6500.20    2627.70     4770.18      2142.48        928.60  

[5 rows x 24 columns]

In [53]:
# Create a column for the total price using the subtotal and the shipping price along with a sales tax of 9.25%
df['Line_price'] = round((df['subtotal'] + df['shipping_price'])*1.0925,2)
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ... unit_cost unit_weight  qty  line_number  \
0          17        2023  ...    762.71        7.50  105            1   
1          20        2023  ...     15.09        1.49   21            0   
2           4        2023  ...      7.86        1.68   39            6   
3          17        2023  ...     24.85        1.23   29            3   
4          21        2023  ...    108.17       46.43   20            1   

    subtotal  shipping_price  line_cost  Line_price  line_profit  total_weight  
0  115164.00         5512.50   85597.05   131839.08     46242.03        787.50  
1     523.95          312.90     629.79      914.26       284.47         31.29  
2     527.28          458.64     765.18     1077.12       311.94         65.52  
3    1056.18          356.70    1077.35     1543.57       466.22         35.67  
4    3902.00         6500.20    8663.60    11364.40      2700.80        928.60  

[5 rows x 24 columns]

In [54]:
# Create a column for the cost of each line using unit cost, qty, and
# shipping price (assume the shipping cost is exactly what is charged to the client).

df['line_cost'] = df['unit_cost'] * df['qty'] + df['shipping_price']
df.head(3)

first     last              job               phone  \
0   Donald  Harding     Immunologist  793-904-7725x39308   
1  Tiffany    Myers  Music therapist    201.442.4543x942   
2  Shannon   Watson     Immunologist   687.737.9424x8503   

                               email  client_id  order_id  order_date  \
0   harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1  myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2              swatson8146@payne.net      57113   1902144  2023-01-29   

   order_week  order_year  ... unit_cost unit_weight  qty  line_number  \
0          17        2023  ...    762.71        7.50  105            1   
1          20        2023  ...     15.09        1.49   21            0   
2           4        2023  ...      7.86        1.68   39            6   

    subtotal  shipping_price  line_cost  Line_price  line_profit  total_weight  
0  115164.00         5512.50   85597.05   131839.08     46242.03        787.50  
1     523.95          312.90     629.79      914.26       284.47         31.29  
2     527.28          458.64     765.18     1077.12       311.94         65.52  

[3 rows x 24 columns]

In [55]:
# Create a column for the profit of each line using line cost and line price
df['line_profit'] = df['Line_price'] - df['line_cost']
df.head(3)


first     last              job               phone  \
0   Donald  Harding     Immunologist  793-904-7725x39308   
1  Tiffany    Myers  Music therapist    201.442.4543x942   
2  Shannon   Watson     Immunologist   687.737.9424x8503   

                               email  client_id  order_id  order_date  \
0   harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1  myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2              swatson8146@payne.net      57113   1902144  2023-01-29   

   order_week  order_year  ... unit_cost unit_weight  qty  line_number  \
0          17        2023  ...    762.71        7.50  105            1   
1          20        2023  ...     15.09        1.49   21            0   
2           4        2023  ...      7.86        1.68   39            6   

    subtotal  shipping_price  line_cost  Line_price  line_profit  total_weight  
0  115164.00         5512.50   85597.05   131839.08     46242.03        787.50  
1     523.95          312.90     629.79      914.26       284.47         31.29  
2     527.28          458.64     765.18     1077.12       311.94         65.52  

[3 rows x 24 columns]

## Part 3: Confirm your work
You have email receipts showing that the total prices for 3 orders. Confirm that your calculations match the receipts. Remember, each order has multiple lines.

Order ID 2742071 had a total price of \$152,811.89

Order ID 2173913 had a total price of \$162,388.71

Order ID 6128929 had a total price of \$923,441.25


In [56]:
# Check your work using the totals above
ids = [2742071, 2173913, 6128929]

for id in ids:
     total_price = df[df['order_id'] == id]['Line_price'].sum()
     print(f"Order ID {id} total price: ${total_price:.2f}")

Order ID 2742071 total price: $152811.89
Order ID 2173913 total price: $162388.71
Order ID 6128929 total price: $923441.25


## Part 4: Summarize and Analyze
Use the new columns with confirmed values to find the following information.

In [59]:
# How much did each of the top 5 clients by quantity spend? Check your work from Part 1 for client ids.
client_spend = {}
for id in top5_clients_list:
    total_price = df[df['client_id'] == id]['Line_price'].sum()
    client_spend[id] = total_price
    print(f"{id}: ${total_price:,.2f}")

    


33615: $8,377,308.52
66037: $10,259,514.79
46820: $9,743,794.36
38378: $12,906,550.87
24741: $82,268,892.02


In [160]:
# Create a summary DataFrame showing the totals for the for the top 5 clients with the following information:
# total units purchased, total shipping price, total revenue, and total profit. 
client_ids = []
total_units_purchased = []
total_shipping_price = []
total_revenue =[]
total_profit = []

for id in top5_clients_list:
    cleint = df[df['client_id'] == id]
    client_ids.append(id)
    total_units_purchased.append(cleint['qty'].sum())
    total_shipping_price.append(cleint['shipping_price'].sum())
    total_revenue.append(cleint['Line_price'].sum())
    total_profit.append(cleint['line_profit'].sum())

summary_df = pd.DataFrame({ 'client id': client_ids, 'Total Units Purchased': total_units_purchased, 'Total Shipping Price': total_shipping_price,
                           'Total Revenue': total_revenue, 'Total Profit': total_profit})
summary_df
    

client id  Total Units Purchased  Total Shipping Price  Total Revenue  \
0      33615                  64313            1828984.89     8377308.52   
1      66037                  43018            1395151.85    10259514.79   
2      46820                  75768            1601448.84     9743794.36   
3      38378                  73667            3429455.40    12906550.87   
4      24741                 239862            5126448.37    82268892.02   

   Total Profit  
0    2201994.61  
1    3255031.81  
2    2736602.72  
3    3271829.89  
4   36579992.31

In [161]:
money_columns = summary_df.drop('client id', axis = 1)

In [162]:
# Format the data and rename the columns to names suitable for presentation.
# Define the money columns. 


# Define a function that converts a dollar amount to millions
def currency_format_millions(val):
    return val / 1_000_000
    
summary_df = summary_df.rename(columns=lambda x: x.replace(' (M) (M)', ' (M)'))
# Apply the currency_format_millions function to the money columns and create new columns for the millions format
for column in money_columns:
     if f'{column} (M)' not in summary_df.columns:
        summary_df[f'{column} (M)'] = summary_df[column].apply(currency_format_millions)



# Rename the columns to reflect the change in the money format
summary_df.rename(columns={
    summary_df.columns[0]: 'Client ID',
    summary_df.columns[1]: 'Total Units Purchased (Mills)',
    summary_df.columns[2]: 'Total Shipping Price (Mills)',
    summary_df.columns[3]: 'Total Revenue (Mills)',
    summary_df.columns[4]: 'Total Profit (Mills)'
}, inplace=True)

# Drop the original columns
summary_df

Client ID  Total Units Purchased (Mills)  Total Shipping Price (Mills)  \
0      33615                          64313                    1828984.89   
1      66037                          43018                    1395151.85   
2      46820                          75768                    1601448.84   
3      38378                          73667                    3429455.40   
4      24741                         239862                    5126448.37   

   Total Revenue (Mills)  Total Profit (Mills)  Total Units Purchased (M)  \
0             8377308.52            2201994.61                   0.064313   
1            10259514.79            3255031.81                   0.043018   
2             9743794.36            2736602.72                   0.075768   
3            12906550.87            3271829.89                   0.073667   
4            82268892.02           36579992.31                   0.239862   

   Total Shipping Price (M)  Total Revenue (M)  Total Profit (M)  
0                  1.828985           8.377309          2.201995  
1                  1.395152          10.259515          3.255032  
2                  1.601449           9.743794          2.736603  
3                  3.429455          12.906551          3.271830  
4                  5.126448          82.268892         36.579992

In [165]:
# Sort the updated data by "Total Profit (millions)" form highest to lowest and assign the sort to a new DatFrame.
sorted_df = summary_df.sort_values(by='Total Profit (Mills)', ascending=False)
sorted_df = summary_df = summary_df.iloc[:, :5]
sorted_df

Client ID  Total Units Purchased (Mills)  Total Shipping Price (Mills)  \
0      33615                          64313                    1828984.89   
1      66037                          43018                    1395151.85   
2      46820                          75768                    1601448.84   
3      38378                          73667                    3429455.40   
4      24741                         239862                    5126448.37   

   Total Revenue (Mills)  Total Profit (Mills)  
0             8377308.52            2201994.61  
1            10259514.79            3255031.81  
2             9743794.36            2736602.72  
3            12906550.87            3271829.89  
4            82268892.02           36579992.31

In [167]:
text = "the top 5 clients appear to have made major contributions to the companys proffit and revenue. Client 24741 Kendra Garrett purchased the most and made to company the most profit. This data shows that these clients have been loyal and the company should mantain these relationships."

In [168]:
print(text)

the top 5 clients appear to have made major contributions to the companys proffit and revenue. Client 24741 Kendra Garrett purchased the most and made to company the most profit. This data shows that these clients have been loyal and the company should mantain these relationships.
